src/usernotifications.cs

```diff
@@ -0,0 +1,653 @@
+		Default,
+		Dismiss
+	}
+
+	[Introduced (PlatformName.iOS, 10, 0)]
+	[Introduced (PlatformName.TvOS, 10, 0)]
+	[Introduced (PlatformName.WatchOS, 3, 0)]
+	[Native]
+	public enum UNAuthorizationStatus : nint {
+		NotDetermined = 0,
+		Denied,
+		Authorized
+	}
+
+	[Introduced (PlatformName.iOS, 10, 0)]
+	[Introduced (PlatformName.TvOS, 10, 0)]
+	[Introduced (PlatformName.WatchOS, 3, 0)]
+	[Native]
+	public enum UNNotificationSetting : nint {
+		NotSupported = 0,
+		Disabled,
+		Enabled
+	}
+
+	[Introduced (PlatformName.iOS, 10, 0)]
+	[Unavailable (PlatformName.TvOS)]
+	[Unavailable (PlatformName.WatchOS)]
+	[Native]
+	public enum UNAlertStyle : nint {
+		None = 0,
+		Banner,
+		Alert
+	}
+
+	[Introduced (PlatformName.iOS, 10, 0)]
+	[Introduced (PlatformName.TvOS, 10, 0)]
+	[Introduced (PlatformName.WatchOS, 3, 0)]
+	[Native]
+	[Flags]
+	public enum UNAuthorizationOptions : nuint {
+		None = 0,
+	    Badge = (1 << 0),
+	    Sound = (1 << 1),
+	    Alert = (1 << 2),
+	    CarPlay = (1 << 3)
+	}
+
+	[Introduced (PlatformName.iOS, 10, 0)]
+	[Introduced (PlatformName.TvOS, 10, 0)]
+	[Introduced (PlatformName.WatchOS, 3, 0)]
+	[Native]
+	[Flags]
+	public enum UNNotificationPresentationOptions : nuint {
+	    None = 0,
+		Badge = (1 << 0),
+	    Sound = (1 << 1),
+	    Alert = (1 << 2)
+	}
+
+	[Category]
+	[BaseType (typeof (NSString))]
+	interface NSString_UNUserNotificationCenterSupport {
+
+		[Introduced (PlatformName.iOS, 10, 0)]
+		[Introduced (PlatformName.WatchOS, 3, 0)]
+		[Unavailable (PlatformName.TvOS)]
+		[Static]
+		[Export ("localizedUserNotificationStringForKey:arguments:")]
+		string GetLocalizedUserNotificationString (string key, [NullAllowed] NSObject [] arguments);
+	}
+
+	[Introduced (PlatformName.iOS, 10, 0)]
+	[Introduced (PlatformName.TvOS, 10, 0)]
+	[Introduced (PlatformName.WatchOS, 3, 0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface UNNotification : NSCopying, NSSecureCoding {
+
+		[Export ("date", ArgumentSemantic.Copy)]
+		NSDate Date { get; }
+
+		 [Export ("request", ArgumentSemantic.Copy)]
+		 UNNotificationRequest Request { get; }
+	}
+
+	[Introduced (PlatformName.iOS, 10, 0)]
+	[Introduced (PlatformName.WatchOS, 3, 0)]
+	[Unavailable (PlatformName.TvOS)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface UNNotificationAction : NSCopying, NSSecureCoding {
+
+		[Export ("identifier")]
+		string Identifier { get; }
+
+		[Export ("title")]
+		string Title { get; }
+
+		[Export ("options", ArgumentSemantic.Assign)]
+		UNNotificationActionOptions Options { get; }
+
+		[Static]
+		[Export ("actionWithIdentifier:title:options:")]
+		UNNotificationAction FromIdentifier (string identifier, string title, UNNotificationActionOptions options);
+	}
+
+	[Introduced (PlatformName.iOS, 10, 0)]
+	[Introduced (PlatformName.WatchOS, 3, 0)]
+	[Unavailable (PlatformName.TvOS)]
+	[BaseType (typeof (UNNotificationAction))]
+	interface UNTextInputNotificationAction {
```

Comment:
the base type does not have a default .ctor - so it looks likely this one should not have one too